### DISCUSSION TITLE
-define vs -set for modulate

### DISCUSSION THREAD

**Discussion by hyperknot** [ImageMagick/ImageMagick#3303] [Help]
The [usage document says](https://legacy.imagemagick.org/Usage/color_mods/#modulate_colorspace) that for modulating in other color spaces I have to use the following command line:
```
-define modulate:colorspace=LCHab
```

Meanwhile [in modulate docs](https://legacy.imagemagick.org/script/command-line-options.php?#modulate), it says:
```
-set option:modulate:colorspace hsb
```

Are these the same? Also why is LCHab not listed on the docs page, or is it the same as one other one?

**Comment by hyperknot** [comment]
I've made a test and they do totally different results. I'm even more confused now.

**Reply by hyperknot** [reply]
Actually not, the files are only different because of some headers. When outputted to TGA both of them generate the same hash. So can I conclude that they are the same?

**Comment by snibgo** [comment]
"-define modulate:colorspace=XX" and "-set option:modulate:colorspace XX", where XX is a colorspace, give the same result.

You can use any 3-channel colorspace you want. [EDIT: well, not really. See below.]

**Reply by hyperknot** [reply]
Thanks! I've written a separate thread about improving modulate's documentation: https://github.com/ImageMagick/ImageMagick/discussions/3305

**Reply by snibgo** [reply]
Sorry, I was wrong. Although you can use any colorspace, the operation is only really for HCL, HCLp, HSB, HSI, HSL, HSV, HWB, LCH, LCHab and LCHuv. If you use any other colorspace, the effect is the same as HSL.
